sys/net/gnrc/network_layer/ipv6/ext/gnrc_ipv6_ext.c

```diff
@@ -36,6 +41,32 @@ bool gnrc_ipv6_ext_demux(kernel_pid_t iface, gnrc_pktsnip_t *pkt,
             case PROTNUM_IPV6_EXT_HOPOPT:
             case PROTNUM_IPV6_EXT_DST:
             case PROTNUM_IPV6_EXT_RH:
+                if ((tmp = gnrc_pktbuf_start_write(pkt)) == NULL) {
+                    DEBUG("ipv6: could not get a copy of pkt\n");
+                    gnrc_pktbuf_release(pkt);
+                }
+                hdr = pkt->data;
+                res = ipv6_ext_rh_process(hdr, (ipv6_ext_rh_t *)ext);
+                if (res == EXT_RH_CODE_ERROR) {
+                    /* TODO: send ICMPv6 error codes */
+                    return false;
+                }
+                else if (res == EXT_RH_CODE_FORWARD) {
+                    /* forward packet */
+                    hdr->src = *gnrc_ipv6_netif_find_best_src_addr(iface, &hdr->dst);
```

Comment:
We should keep the source address as is. RFC 6554 Section 2:
"S's address is carried in the IPv6 header's Source Address field." where S is the originating node.